### PR TITLE
Adding metadata to caseevents

### PR DESCRIPTION
**Pull request by ryangrundy7**
# Motivation and Context
In response-operation-ui, we want to be able to see the respondent name on the responses status page when a external user has completed a collection exercise. For this to happen, a new column needs to be added to the caseevents table which will hold metadata required to get the respondent name.

# What has changed
- Added a new column to case events table

# How to test?
- Run `mvn clean install` and it should build successfully
- Run with the [case-svc-api PR](https://github.com/ONSdigital/rm-casesvc-api/pull/33), [R-ops branch](https://github.com/ONSdigital/response-operations-ui/tree/us211-display-respondent-for-seft-rus-ce-in-response-status), [Frontstage branch](https://github.com/ONSdigital/ras-frontstage/tree/us211-display-respondent-for-seft-rus-ce-in-response-status) and complete a SEFT collection exercise. When completed, look on response ops response status page and the completed collex should have a respondent name attached to it.
- Run with acceptance tests

# Links
[Trello](https://trello.com/c/1rcfmtok)

### CODE REVIEW

File: src/test/java/uk/gov/ons/ctp/response/casesvc/endpoint/CaseEndpointIT.java [hunk 1]
```diff
@@ -41,6 +42,7 @@
 public class CaseEndpointIT {
 
   private UUID collectionExerciseId;
+  private static HashMap<String, String> metadata;
```
Comment (mfcrocker) [inline src/test/java/uk/gov/ons/ctp/response/casesvc/endpoint/CaseEndpointIT.java]: Does this need to be `static`?

Comment (ryangrundy7) [reply src/test/java/uk/gov/ons/ctp/response/casesvc/endpoint/CaseEndpointIT.java]: No it doesn't need to be static 👍

Comment (NickGrantONS) [reply src/test/java/uk/gov/ons/ctp/response/casesvc/endpoint/CaseEndpointIT.java]: This should also be a `Map` not a `HashMap`

File: pom.xml [hunk 2]
```diff
@@ -158,7 +158,7 @@
         <dependency>
             <groupId>uk.gov.ons.ctp.product</groupId>
             <artifactId>casesvc-api</artifactId>
-            <version>10.49.30</version>
+            <version>10.49.33-SNAPSHOT</version>
```
Comment (NickGrantONS) [inline pom.xml]: Don't forget to update this

File: src/main/java/uk/gov/ons/ctp/response/casesvc/domain/model/CaseEvent.java [hunk 3]
```diff
@@ -60,4 +66,8 @@
 
   @Column(name = "subcategory")
   private String subCategory;
+
+  @Column(name = "metadata")
+  @Type(type = "jsonb")
```
[the remark this answers is not in the document]
Comment (NickGrantONS) [reply src/main/java/uk/gov/ons/ctp/response/casesvc/domain/model/CaseEvent.java]: I have some concerns about what might happen if we use more than key-value pairs. Did you try it with an array of integers, for example, like this: `[1,2,3,4,5]`

What about a more complex type like this `{"foo":"bar","disco":[1,2,3,4,5]}`

**Comment by NickGrantONS** [comment]
We should also enhance the Case service so that it stores the `partyId` metadata in `CaseReceiptReceiver` so that we have the metadata stored telling us who _completed_ the EQ

File: src/main/java/uk/gov/ons/ctp/response/casesvc/domain/model/CaseEvent.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (NickGrantONS) [reply src/main/java/uk/gov/ons/ctp/response/casesvc/domain/model/CaseEvent.java]: actually maybe ignore me... I guess there's only one way that this stuff can get into the database in the first place

File: src/test/java/uk/gov/ons/ctp/response/casesvc/endpoint/CaseEndpointUnitTest.java [hunk 4]
```diff
@@ -804,6 +815,39 @@ public void createCaseEventGoodJson() throws Exception {
...
+    metadata.put("partyId", CASE_EVENT_PARTY_UUID);
+    caseEventDTO.setMetadata(metadata);
+    caseEventDTO.setCreatedBy("Test");
+    caseEndpoint.createCaseEvent(CASE9_ID, caseEventDTO, bindingResult);
+
+    // Then
+    ArgumentCaptor<CaseEvent> caseEventArgumentCaptor = ArgumentCaptor.forClass(CaseEvent.class);
+    verify(caseService).createCaseEvent(caseEventArgumentCaptor.capture(), any(Case.class));
+    assertEquals(
+        CASE_EVENT_PARTY_UUID, caseEventArgumentCaptor.getValue().getMetadata().get("partyId"));
```
Comment (NickGrantONS) [inline src/test/java/uk/gov/ons/ctp/response/casesvc/endpoint/CaseEndpointUnitTest.java]: 👍

**Pull request merged**